backend/gke/force-clean-cluster.sh

```diff
@@ -49,3 +56,27 @@ for disk in $(jq -r ".disks[] | select(${DISK_SELECTOR:-true}) | .name" "${RESOU
     ${PREFIX} gcloud compute disks delete "${disk}" \
         --quiet --project="${GKE_PROJECT}" --zone="${GKE_LOCATION}"
 done
+
+dns_zone="$(jq -r '.dns.zone' "${RESOURCE_LIST}")"
+mapfile -t recordsets < <(
+    jq -r --arg zone "${dns_zone}" '
+        . as $root
+        | .dns.entries[]
+        | select([.info["external-dns/owner"]] | inside($root.clusters | map(.name)) | not)
+        | .recordsets[]
+        | ["--name=\(.name)", "--type=\(.type)", "--zone=\($zone)", "--ttl=\(.ttl)"] + .rrdatas
+        | @sh
+    ' "${RESOURCE_LIST}"
+)
+
+if [[ "${#recordsets[@]}" -gt 0 ]]; then
+    ${PREFIX} gcloud dns record-sets transaction start --zone="${dns_zone}"
+    for recordset in "${recordsets[@]}"; do
+        record_name="$(grep --perl-regexp --only-matching -- "--name='?\K[^']+(?=')" <<<"${recordset}")"
+        record_type="$(grep --perl-regexp --only-matching -- "--type='?\K[^']+(?=')" <<<"${recordset}")"
+        info "Removing DNS record set ${record_name} (type ${record_type})"
+        eval "${PREFIX} gcloud dns record-sets transaction remove ${recordset}"
+    done
+    info "Committing DNS changes…"
+    ${PREFIX} gcloud dns record-sets transaction execute --zone="${dns_zone}"
```

Comment:
I wonder if this is atomic, or if we would get into trouble when several jobs try to execute a transaction.